src/psyclone/dynamo0p3.py

```diff
@@ -6815,14 +7064,32 @@ def _upper_bound_fortran(self):
             # determines the iteration space of this kernel and that
             # is set-up to be the one on the coarse mesh (in
             # DynKernelArguments.iteration_space_arg()).
-            mesh_name = "mesh_" + self._field_name
+            tag_name = "mesh_" + self._field_name
         else:
             # It's not an inter-grid kernel so there's only one mesh
-            mesh_name = "mesh"
-        # Use InvokeSchedule SymbolTable to share the same symbol for all
-        # Loops in the Invoke.
-        mesh = self.ancestor(InvokeSchedule).symbol_table.\
-            find_or_create_tag(mesh_name).name
+            tag_name = "mesh"
+
+        # The symbol for the mesh will already have been added to the
+        # symbol table associated with the InvokeSchedule.
+        return self.ancestor(InvokeSchedule).symbol_table.\
+            lookup_with_tag(tag_name).name
+
+    def _upper_bound_fortran(self):
+        ''' Create the Fortran code that gives the appropriate upper bound
+        value for this type of loop.
+
+        TODO: Issue #440. upper_bound_fortran should generate PSyIR.
+
+        :return: Fortran code for the upper bound of this loop
```

Comment:
Full stop?

Reply:
Perhaps use `returns` as most of your changes use this.

Reply:
Yes and done.